### PR TITLE
Automatically attempt to reconnect websockets

### PR DESCRIPTION
**Pull request by eliperelman**
Does the reconnecting logic (while basic) seem sane to you two?

### CODE REVIEW

**Comment by eliperelman** [comment]
> Upon connecting, start an interval to check if the websocket exists and has been closed. If so, reconnect the websocket.

File: src/WebListener.js [hunk 1]
```diff
@@ -134,20 +141,23 @@ export default class WebListener {
 
   handleError = () => this.emitter.emit('error', new Error('WebSocket error'));
 
-  handleClose = () => this.emitter.emit('close');
+  handleClose = () => {
+    this.emitter.emit('close');
+    this.reconnect();
```
Comment (jonasfj) [inline src/WebListener.js]: So it'll emit close before reconnecting?

File: src/WebListener.js [hunk 2]
```diff
@@ -53,26 +45,41 @@ export default class WebListener {
...
+          }
+        }, this.options.reconnectInterval);
+
+        return isReady;
+      });
+  }
+
+  reconnect() {
+    if (!this.socket || this.socket.readyState === WebSocket.CLOSED) {
+      this.connect();
```
Comment (jonasfj) [inline src/WebListener.js]: I would suggest emitting 'reconnected' after a successful reconnect...

In task inspector we should ideally reload and synchronize state when reconnect is emitted...

File: src/WebListener.js (hunk 1 above)
Comment (eliperelman) [reply src/WebListener.js]: Yes, we still will technically receive a close message prior to reconnecting.

File: src/WebListener.js (hunk 2 above)
Comment (eliperelman) [reply src/WebListener.js]: I'm not too keen on doing a full reload, but I'm not sure we have much of a choice.

**Comment by eliperelman** [comment]
I've updated the commit with some content about the WebListener. Give it a read and if everything seems kosher with you I'll merge.

**Comment by eliperelman** [comment]
Released in v4.0.0.

**Pull request merged**